lib/pure/dynlib.nim

```diff
@@ -137,6 +137,28 @@ when defined(posix):
   proc symAddr(lib: LibHandle, name: cstring): pointer =
     return dlsym(lib, name)
 
+elif defined(nintendoswitch):
+  #
+  # =========================================================================
+  # Nintendo switch DevkitPro sdk does not have these. Raise an error if called.
+  # =========================================================================
+  #
+
+  proc dlclose(lib: LibHandle) =
+    raise newException(OSError, "dlclose not implemented on Nintendo Switch!")
```

Comment:
It would be better if these would cause a compile-time error but I can imagine why that's rather hard to do...

Reply:
Is there a way that you usually do this in Nim?

Reply:
``proc dlclose(lib: LibHandle) {.error: "not implemented".}`` but it's a compilerProc so it needs a patch in ``cgen.nim``.